### PR TITLE
Authentication comms

### PR DESCRIPTION
**Pull request by sturoscy-personal**
This should resolve the authentication communication ticket. As always, open to suggestions :)

### CODE REVIEW

File: src/components/Endpoint.tsx [hunk 1]
```diff
@@ -143,7 +145,7 @@ const Endpoint = (props) => {
 
         if (config.is_hub) {
           setAPIError(null);
-          setLoginURL(error_response.login_url);
+          setHubResponse(error_response);
```
Comment (sturoscy-personal) [inline src/components/Endpoint.tsx:148]: This is probably the biggest change. I'm passing the entire error response from the operation_ls failure. With this change, the hub login widget had to be refactored to handle the response.

File: src/components/HubLoginWidget.tsx [hunk 2]
```diff
@@ -63,20 +89,54 @@ export const HubLogin = (props) => {
             </div>
           )}
 
+          {props.endpoint ? (
```
[the remark this answers is not in the document]
Comment (NickolausDS) [reply src/components/HubLoginWidget.tsx:92]: If I understand this right, if `endpoint` isn't set it assumes the user is logging in for the first time?

**Comment by NickolausDS** [comment]
Tests are failing due to #93 and can be ignored

**Comment by NickolausDS** [comment]
New changes look good, this is what I got when simulating the hub locally: 
Screen Shot 2022-07-08 at 10 40 23 AM
Although, I think it was showing the error message twice here: 
Screen Shot 2022-07-08 at 10 42 18 AM

File: src/components/HubLoginWidget.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sturoscy-personal) [reply src/components/HubLoginWidget.tsx:92]: No this is just checking if the prop exists. If it does, display the endpoint info in the auth/login widget.

**Comment by sturoscy-personal** [comment]
> New changes look good, this is what I got when simulating the hub locally: Screen Shot 2022-07-08 at 10 40 23 AM Although, I think it was showing the error message twice here: Screen Shot 2022-07-08 at 10 42 18 AM

Oh weird! I hadn't encountered that bug. I'll take a look.

**Comment by NickolausDS** [comment]
Dependent on #92

**Comment by NickolausDS** [comment]
> Oh weird! I hadn't encountered that bug. I'll take a look.

Do you have a fix for this? Otherwise, the PR looks good to me.

**Comment by sturoscy-personal** [comment]
> > Oh weird! I hadn't encountered that bug. I'll take a look.
> 
> Do you have a fix for this? Otherwise, the PR looks good to me.

Shoot not yet. I will take a look this afternoon. Weird that its duplicating the error.

**Review by NickolausDS — APPROVED** [review]
Looks good to me!

**Pull request merged**